src/controllers/error.controller.ts

```diff
@@ -13,7 +14,11 @@ const pageNotFound = (req: Request, res: Response) => {
  */
 const errorHandler = (err: Error, req: Request, res: Response, _next: NextFunction) => {
   logger.errorRequest(req, `An error has occurred. Re-routing to the error screen - ${err.stack}`);
-  res.status(500).render(config.ERROR_PAGE);
+  const userEmail = getLoggedInUserEmail(req.session);
```

Comment:
This controller should be the simplest possible, if an error occurs on `getLoggedInUserEmail(req.session);` there is nothing that will catch it and safely return. So the sign out link should be removed from the error page.
Do you agree @markpit

Reply:
what would the error page look like if we rendered it without the user email? We could wrap the getLoggedInUserEmail line in a try catch and just log an error but render the page anyway without the email?
otherwise, i'd say remove logout from the error page

Reply:
So what's the consensus, add the wrapper or just remove?

Reply:
If you wrap it and catch it, we are going to sendup in the same point. I suggest to remove

Reply:
ok remove sounds best

Reply:
Just the one change, we still require the layout html to be inherited and the sign out banner is controlled by the presence of a userEmail, so have decided to achieve the removal by making the userEmail falsy.

Reply:
An alternative might be to introduce a new flag variable such as "excludeSignOutBanner" and add it to the conditions so if "userEmail and excludeSignOutBanner == falsy..."

Reply:
if you render the page without a userEmail then that should be falsy too, ie. no change needed to the controller
 res.status(500).render(config.ERROR_PAGE);

Reply:
The difficulty is that when we first go to the error page it is not falsy so something has to be done to ensure that it will be at that time as well.